muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/SyncStudentEventHandler.java

```diff
@@ -30,16 +37,30 @@
   @Inject
   private ChatSyncController chatSyncController;
   
+  @Inject
+  private ChatController chatController;
+  
   @Inject
   private WorkspaceEntityController workspaceEntityController;
   
+  @Inject
+  private WorkspaceUserEntityController workspaceUserEntityController;
+  
   public synchronized void onSchoolDataWorkspaceUserDiscoveredEvent(@Observes SchoolDataWorkspaceUserDiscoveredEvent event) {
     //kun kurssille pelmahtaa uusi opiskelija
-    
     String userIdentifier = event.getUserIdentifier();
     String userDataSource = event.getUserDataSource();
     SchoolDataIdentifier user = new SchoolDataIdentifier(userIdentifier, userDataSource);
-    chatSyncController.syncStudent(user);
+    String workspaceDataSource = event.getWorkspaceDataSource();
+    String workspaceIdentifier = event.getWorkspaceIdentifier();
+    
+    WorkspaceEntity workspace = workspaceEntityController.findWorkspaceByDataSourceAndIdentifier(workspaceDataSource, workspaceIdentifier);
+    
+    WorkspaceChatSettings workspaceChatStatus = chatController.findWorkspaceChatSettings(workspace.getId());
+
+    if (workspaceChatStatus.getStatus() != null && workspaceChatStatus.getStatus() == WorkspaceChatStatus.ENABLED) {
```

Comment:
Instead of `workspaceChatStatus.getStatus() != null` use `workspaceChatStatus != null` to check that the chat status object was returned from database.